runtime/src/main/scala/akka/grpc/scaladsl/scaladsl.scala

```diff
@@ -0,0 +1,35 @@
+package akka.grpc
+
+import java.util.Base64
+
+import akka.http.scaladsl.model.HttpHeader
+import akka.util.ByteString
+
+import scala.collection.immutable
+
+package object scaladsl {
+  type MetadataMap = immutable.Map[String, Seq[MetadataEntry]]
```

Comment:
The existing `Metadata` facade is a super limited interface that is inadequate for "real" use by service implementers. It only provides `getText()` and `getBinary()` methods returning optional values if given keys exist. It doesn't provide a means for getting all the keys nor for getting multi-valued keys. So some kind of richer interface is needed, and per discussion on #179 I kept the new-code surface small by using a `Map` and reusing `MetadataEntry`. If you want the latter to be kept internal, how would you suggest providing this richer API?

Reply:
Instead of exposing `MetadataEntry` we could also consider adding additional methods to the `Metadata` API. I don't have a strong preference at this point, though.

Reply:
Keeping a simple facade gives us more freedom to change implementation though and optimize for performance etc. I think having the facade makes sense, having an explicit `Metadata.toSeq` to get the collection capabilities for example would perhaps allow for the more advanced use cases?